lgc/patch/PatchBufferOp.cpp

```diff
@@ -513,9 +513,13 @@ void PatchBufferOp::visitGetElementPtrInst(GetElementPtrInst &getElemPtrInst) {
   Value *newGetElemPtr = nullptr;
 
   if (getElemPtrInst.isInBounds())
-    newGetElemPtr = m_builder->CreateInBoundsGEP(m_replacementMap[pointer].second, indices);
+    newGetElemPtr = m_builder->CreateInBoundsGEP(
+        m_replacementMap[pointer].second->getType()->getScalarType()->getPointerElementType(),
```

Comment:
Can you create a local variable for `m_replacementMap[pointer].second` and it's type? This will make the code less repetitive and easier to debug.